src/Microsoft.Windows.ComputeVirtualization/HcsFunctions.cs

```diff
@@ -5,36 +5,112 @@
+        public static extern int HcsGetComputeSystemProperties(IntPtr computeSystem, string propertyQuery, [MarshalAs(UnmanagedType.LPWStr)] out string properties, [MarshalAs(UnmanagedType.LPWStr)] out string result);
 
-        [DllImport("vmcompute.dll", PreserveSig = false, ExactSpelling = true)]
-        public static extern void HNSCall(string method, string path, string request, [MarshalAs(UnmanagedType.LPWStr)] out string response);
+        [DllImport("vmcompute.dll", ExactSpelling = true)]
+        public static extern int HcsModifyComputeSystem(IntPtr computeSystem, string configuration, [MarshalAs(UnmanagedType.LPWStr)] out string result);
+
+        [DllImport("vmcompute.dll", ExactSpelling = true)]
+        public static extern int HcsRegisterComputeSystemCallback(IntPtr computeSystem, NotificationCallback callback, IntPtr context, out IntPtr callbackHandle);
+
+        [DllImport("vmcompute.dll", ExactSpelling = true)]
+        public static extern int HcsUnregisterComputeSystemCallback(IntPtr callbackHandle);
+
+        [DllImport("vmcompute.dll", ExactSpelling = true)]
+        public static extern int HcsCreateProcess(IntPtr computeSystem, string processParameters, out HCS_PROCESS_INFORMATION processInformation, out IntPtr process, [MarshalAs(UnmanagedType.LPWStr)] out string result);
+
+        [DllImport("vmcompute.dll", ExactSpelling = true)]
+        public static extern int HcsOpenProcess(IntPtr computeSystem, uint processId, out IntPtr process, [MarshalAs(UnmanagedType.LPWStr)] out string result);
+
+        [DllImport("vmcompute.dll", ExactSpelling = true)]
+        public static extern int HcsCloseProcess(IntPtr process);
+
+        [DllImport("vmcompute.dll", ExactSpelling = true)]
+        public static extern int HcsTerminateProcess(IntPtr process, [MarshalAs(UnmanagedType.LPWStr)] out string result);
+
+        [DllImport("vmcompute.dll", ExactSpelling = true)]
+        public static extern int HcsGetProcessInfo(IntPtr process, out HCS_PROCESS_INFORMATION processInformation, [MarshalAs(UnmanagedType.LPWStr)] out string result);
+
+        [DllImport("vmcompute.dll", ExactSpelling = true)]
+        public static extern int HcsGetProcessProperties(IntPtr process, [MarshalAs(UnmanagedType.LPWStr)] out string properties, [MarshalAs(UnmanagedType.LPWStr)] out string result);
+
+        [DllImport("vmcompute.dll", ExactSpelling = true)]
+        public static extern int HcsModifyProcess(IntPtr process, string settings, [MarshalAs(UnmanagedType.LPWStr)] out string result);
+
+        [DllImport("vmcompute.dll", ExactSpelling = true)]
+        public static extern int HcsRegisterProcessCallback(IntPtr process, NotificationCallback callback, IntPtr context, out IntPtr callbackHandle);
+
+        [DllImport("vmcompute.dll", ExactSpelling = true)]
+        public static extern int HcsUnregisterProcessCallback(IntPtr callbackHandle);
+
+        public static bool ProcessHcsCall(int resultCode, string result)
```

Comment:
It almost makes sense to me I think to make `ProcessHcsCall` be `HcsException.ProccessResult` more like you did the `Failed` method.